### PR TITLE
converting pvc extension types to dynamic-plugin-sdk

### PR DESCRIPTION
**Pull request by glekner**
- Fix issues related to loading plugins via SERVER_FLAGS.consolePlugins
- dp-sdk-pvc

### CODE REVIEW

**Comment by glekner** [comment]
blocked by #7912

**Comment by glekner** [comment]
can you review @rawagner @vojtechszocs

File: frontend/packages/kubevirt-plugin/console-extensions.json
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/console-extensions.json:81]: Do we need `"required": ["KUBEVIRT"]` here?

File: frontend/packages/kubevirt-plugin/package.json
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/package.json:47]:
```suggestion
      "createVM": "src/components/create-vm.tsx",
```

File: frontend/public/components/persistent-volume-claim.jsx
Comment (vojtechszocs) [inline frontend/public/components/persistent-volume-claim.jsx:199]: Please use extension's `uid` property value for React `key` prop.

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

File: frontend/public/components/persistent-volume-claim.jsx
Comment (christianvogt) [inline frontend/public/components/persistent-volume-claim.jsx:363]: Why the change to `useResolvedExtensions`? Doesn't look like this requires resolving.

File: frontend/packages/kubevirt-plugin/src/components/dv-status/dv-import-status.tsx
Comment (christianvogt) [inline frontend/packages/kubevirt-plugin/src/components/dv-status/dv-import-status.tsx:19]: localize
Seems like the kubevirt plugin has a lot of missing localization.

File: frontend/packages/kubevirt-plugin/console-extensions.json
Comment (christianvogt) [inline frontend/packages/kubevirt-plugin/console-extensions.json:89]: localize using `%...%`

File: frontend/packages/kubevirt-plugin/console-extensions.json
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/console-extensions.json:89]:
```suggestion
      "label": "%kubevirt-plugin~With Data upload form%",
```

File: frontend/public/components/persistent-volume-claim.jsx
Comment (vojtechszocs) [inline frontend/public/components/persistent-volume-claim.jsx:67]: `useExtensions` hook result is the extensions array itself, so no need for `[ .. ]` destructuring.

`useResolvedExtensions` hook result, on the other hand, is a tuple `[resolvedExtensions, resolved, errors]` so that's a good case for destructuring.

Comment (vojtechszocs) [reply frontend/public/components/persistent-volume-claim.jsx:67]:
```suggestion
  const pvcStatusExtensions = useExtensions(isPVCStatus);
```

File: frontend/packages/kubevirt-plugin/console-extensions.json
Comment (vojtechszocs) [reply frontend/packages/kubevirt-plugin/console-extensions.json:89]: You might also want to update `locales/en/kubevirt-plugin.json` file.

File: frontend/packages/kubevirt-plugin/package.json
Comment (vojtechszocs) [reply frontend/packages/kubevirt-plugin/package.json:47]: Can be done later.

File: frontend/public/components/persistent-volume-claim.jsx
Comment (vojtechszocs) [inline frontend/public/components/persistent-volume-claim.jsx:15]: Once #8224 is merged this should be simply import from `@console/dynamic-plugin-sdk`.

File: frontend/public/components/persistent-volume-claim.jsx
Comment (vojtechszocs) [inline frontend/public/components/persistent-volume-claim.jsx:73]: The `status` component is loaded via `CodeRef` so we need to use `useResolvedExtensions` (?)

File: frontend/public/locales/en/public.json
Comment (vojtechszocs) [inline frontend/public/locales/en/public.json:725]: This i18n msg should be scoped under `kubevirt-plugin`.

**Comment by vojtechszocs** [comment]
Rebased on #8156 and resolved merge conflicts.

**Pull request merged**
